src/UploadedFileFactoryInterface.php

```diff
@@ -26,9 +26,9 @@ interface UploadedFileFactoryInterface
      */
     public function createUploadedFile(
         StreamInterface $stream,
-        int $size = null,
+        ?int $size = null,
         int $error = \UPLOAD_ERR_OK,
-        string $clientFilename = null,
-        string $clientMediaType = null
+        ?string $clientFilename = null,
+        ?string $clientMediaType = null
```

Comment:
> Note: This syntax is supported as of PHP 7.1.0, and predates generalized union types support. 

As this package supports PHP 7.0, we cannot use nullable types without other changes.

Additionally, such changes MUST be targeted to the PHP-FIG specification _before_ being targeted to the implementation. cc @mbniebergall @Jean85 @weierophinney

Reply:
Thanks for the quick review. You are right, we can't make these changes without bumping the minimum PHP version to 7.1+. 

As far as specification goes, I think this is merely an implementation detail. Both `string $clientFilename = null` and `?string $clientFilename = null` are effectively identical for LSP and Reflection APIs, so changing this should have no impact on PSR implementations.

Reply:
@shadowhand `int $paramName = null` is semantically equivalent to `?int $paramName`, and it's been that way since parameter type hints were introduced; PHP just didn't formalize the concept of nullable types until 7.1.

I think at this point, it's fine to bump to 7.1. When you look at Packagist stats, across all packages, 7.0 only has a 1.3% usage... and when you look at the [http-factory stats](https://packagist.org/packages/psr/http-factory/php-stats)... it has **0%** usage. I think we can safely bump.